#! /bin/sh /usr/share/dpatch/dpatch-run
## 202_gem_default_dir.dpatch by akira yamada <akira@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Adjust the default RubyGems directory.

@DPATCH@
Index: b/lib/rubygems/defaults.rb
===================================================================
--- a/lib/rubygems/defaults.rb	2010-07-20 21:45:46.000000000 +0900
+++ b/lib/rubygems/defaults.rb	2010-07-20 21:46:13.000000000 +0900
@@ -15,13 +15,17 @@
   ##
   # Default home directory path to be used if an alternate value is not
   # specified in the environment
-
+  #
+  # Debian patch: search order of this directory.
+  #   1. GEM_HOME enviroment variable
+  #      (Using this, Gems are to be installed in any path as you like)
+  #   2. /var/lib/gems/{ruby version} (This is the default path in Debian system)
   def self.default_dir
     if defined? RUBY_FRAMEWORK_VERSION then
       File.join File.dirname(ConfigMap[:sitedir]), 'Gems',
                 ConfigMap[:ruby_version]
     else
-      ConfigMap[:sitelibdir].sub(%r'/site_ruby/(?=[^/]+)', '/gems/')
+      File.join('/var/lib', 'gems', ConfigMap[:ruby_version])
     end
   end
 
